src/libraries/System.Net.Sockets/src/System/Net/Connections/SocketsConnectionFactory.cs

```diff
@@ -0,0 +1,159 @@
+        /// </summary>
+        /// <param name="addressFamily">The <see cref="AddressFamily"/> to forward to the socket.</param>
+        /// <param name="socketType">The <see cref="SocketType"/> to forward to the socket.</param>
+        /// <param name="protocolType">The <see cref="ProtocolType"/> to forward to the socket.</param>
+        public SocketsConnectionFactory(
+            AddressFamily addressFamily,
+            SocketType socketType,
+            ProtocolType protocolType)
+        {
+            _addressFamily = addressFamily;
+            _socketType = socketType;
+            _protocolType = protocolType;
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="SocketsConnectionFactory"/> class
+        /// that will forward <see cref="AddressFamily.InterNetworkV6"/> to the Socket constructor.
+        /// </summary>
+        /// <param name="socketType">The <see cref="SocketType"/> to forward to the socket.</param>
+        /// <param name="protocolType">The <see cref="ProtocolType"/> to forward to the socket.</param>
+        /// <remarks>The created socket will be an IPv6 socket with <see cref="Socket.DualMode"/> enabled.</remarks>
+        public SocketsConnectionFactory(SocketType socketType, ProtocolType protocolType)
+            : this(AddressFamily.InterNetworkV6, socketType, protocolType)
+        {
+        }
+
+        /// <inheritdoc />
+        public override async ValueTask<Connection> ConnectAsync(
+            EndPoint? endPoint,
+            IConnectionProperties? options = null,
+            CancellationToken cancellationToken = default)
+        {
+            if (endPoint == null) throw new ArgumentNullException(nameof(endPoint));
+
+            Socket socket = CreateSocket(_addressFamily, _socketType, _protocolType, endPoint, options);
+
+            try
+            {
+                using var args = new TaskSocketAsyncEventArgs();
+                args.RemoteEndPoint = endPoint;
+
+                if (socket.ConnectAsync(args))
+                {
+                    using (cancellationToken.UnsafeRegister(o => Socket.CancelConnectAsync((SocketAsyncEventArgs)o!), args))
+                    {
+                        await args.Task.ConfigureAwait(false);
+                    }
+                }
+
+                if (args.SocketError != SocketError.Success)
+                {
+                    SocketException ex = new SocketException((int)args.SocketError);
+                    if (args.SocketError == SocketError.OperationAborted && cancellationToken.IsCancellationRequested)
+                    {
+                        throw new TaskCanceledException(s_cancellationMessage, ex, cancellationToken);
+                    }
+
+                    throw NetworkErrorHelper.MapSocketException(ex);
+                }
+
+                return new SocketConnection(socket, this, options);
```

Comment:
The options passed to `ConnectAsync` are the user's settings to control `ConnectAsync` behavior.

They should not be inherited by `SocketConnection`.

Reply:
`SocketConnection` needs to have a reference to the settings so it can pass them to `SocketsConnectionFactory`.

Doesn't have to utilize them in `TryGet` as it does currently though.